src/JsonRpcEngine.ts

```diff
@@ -0,0 +1,350 @@
+import SafeEventEmitter from '@metamask/safe-event-emitter';
+import { errorCodes, EthereumRpcError, serializeError } from 'eth-rpc-errors';
+
+type Maybe<T> = T extends Record<string, unknown> ? Partial<T> : T | null | undefined;
```

Comment:
unfortunately, can't use `Record<string, unknown>` here.

when this is `Object` or `Record<string, any>`:

![image](https://user-images.githubusercontent.com/4448075/99570417-85694000-2997-11eb-985e-471ab28b4e11.png)

when this is `Record<string, unknown>`:
![image](https://user-images.githubusercontent.com/4448075/99570533-b0539400-2997-11eb-8f6c-567f49cb5775.png)

the `Partial` isn't applied here.

---
HOWEVER! I don't think we need this `?` at all. just `Partial<T> | null | undefined`
I tested in Playground and even in the event that T is not a record this works, Partial already skips base types that cannot be partial.

Reply:
Ah, I switched to `Record` because our lint rules forbid `Object`. That's great news though! I'll skip the `?`.